README.md

```diff
@@ -14,91 +14,132 @@ another Hadoop file system that executes it against the underlying object store.
 #### Pre-requisites 
 - Install maven 
 
-#### Build instructions
+#### Steps
 
-1. Clone the repo with 
+1. Clone the repo:
     ```shell
     git clone git@github.com:treeverse/hadoop-router-fs.git
     ```
 
-2. Build with maven
-```shell
-mvn clean install
-```
+2. Build with maven:
+   ```shell
+   mvn clean install
+   ```
 
 ## How to configure RouterFS 
 
 ### Configure Spark to use RouterFS
 
-You should tell Spark to use RouterFS as the file system for URIs you would like to transform at runtime. To do that, 
-add the following Spark property to your Spark configurations: 
+You should instruct Spark to use RouterFS as the file system implementation for the URIs you would like to transform at runtime. To do that, 
+add the following property to your Spark configurations: 
 ```properties
 fs.${fromFsScheme}.impl=RouterFileSystem
 ```
 
-For example, by adding the `fs.s3a.impl=RouterFileSystem` you are telling Spark to use RouterFS as the file system for any 
+For example, by adding the `fs.s3a.impl=RouterFileSystem` you are instructing Spark to use RouterFS as the file system for any 
 URI with `scheme=s3a`.
 
-### Add custom Mapping Configurations
+### Add custom mapping configurations
 
 RouterFS consumes your mapping configurations to understand which paths it needs to modify and how to modify them. It then 
-does a simple prefix replacement accordingly. 
-Mapping configurations are Spark properties of the following form:
-`routerfs.mapping.${fromFsScheme}.${mappingIdx}.${replace/with}=${path-prefix}`
-
-Given a URI, RouterFS looks at the mapping configurations defined for the URI scheme, searches for the first mapping
+performs a simple prefix replacement accordingly.  
+Mapping configurations are Hadoop properties of the following form:
+`routerfs.mapping.${fromFsScheme}.${mappingIdx}.(replace|with)=${path-prefix}`  
+For a given URI, RouterFS scans the mapping configurations defined for the URI's scheme, searches for the first mapping
 configuration that matches the URI prefix, and transforms the URI according to the matching configuration.
-For example, given the URI `s3a://bucket/dir1/foo.parquet`, and the following mapping configurations:
+
+#### Notes about mapping configurations:
+* Make sure your source prefix ends with a slash when needed.
```

Comment:
```suggestion

* Make sure your source prefix ends with a slash when needed.
```

Markdown pretty much requires a blank line around header lines.  In practice most implementations including GitHub's work without it, but it may still be easier with some tools.